### PR TITLE
sign convention docs and proof

### PR DESCRIPTION
**Pull request by bradcarman**
Adding documentation about the applied sign convention.  No changes were made to the current library except to add Constant input sources which aligned with the already available input sources.  This PR replaces the previous incorrect PR #246.

### CODE REVIEW

**Comment by baggepinnen** [comment]
something appears off with this PR, it looks like every single file has been modified by the formatter?

**Comment by bradcarman** [comment]
I truly modified about 15 files.  Formatter, maybe a new update, modified a few others.

**Comment by bradcarman** [comment]
Ah OK, makes sense.  I removed the ConstantX components and define them simply in the "sign_convention.md" doc for illustration purposes only.

File: docs/src/connectors/sign_convention.md [hunk 1]
```diff
@@ -0,0 +1,189 @@
+# Sign Convention
+
+A sign convention is recommended for this library that implements the following rule:
+
+> An input component that specifies the *through* variable should be such that an oppossite equality is written such that when connected to a conserved quantity component (i.e. a mass or capacitor component) a positive value for a flow variable represents the accumulation of that conserved quantify over time in the component.
```
Comment (ven-k) [inline docs/src/connectors/sign_convention.md:5]: Minor typo:
```suggestion
> An input component that specifies the *through* variable should be such that an oppossite equality is written such that when connected to a conserved quantity component (i.e. a mass or capacitor component) a positive value for a flow variable represents the accumulation of that conserved quantity over time in the component.
```

File: docs/src/connectors/sign_convention.md [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+In this case we know from Newton that mass times acceleration equals force, therefore the direction of movement is in the opposite direction of the force.  In other words, if we push the mass from left to right (i.e. in the positive direction), then the mass will generate a force in the negative direction.  This would be the general rule for a single port component of any domain.  The exception is for a *through* variable **input** component, which should align the component and port connection *through* variables.  For example, a force input diagram would look like this:
+
+![force](force_input.svg)
+
+For a 2 port connection component, then the *through* variable is exiting each connection port in opposing directions.  Using a damper as an example, if the positive direction is to the right, then the force of the damper is pushing left (i.e. in the negative direction) on connection port `a` and right (positive direction) on connection port `b`.
+
+![damper](damper.svg) -->
+# Discussion
+
+The energy disipation equation that governs the acausal connection definitions should be such that a positive *through* variable input will lead to an increasing *across* variable value.
```
Comment (ven-k) [inline docs/src/connectors/sign_convention.md:48]: Minor typo:
```suggestion
The energy dissipation equation that governs the acausal connection definitions should be such that a positive *through* variable input will lead to an increasing *across* variable value.
```

**Review by ven-k — APPROVED** [review]
This looks great!

**Review by ChrisRackauckas — APPROVED** [review]

**Pull request merged**